test/unit/openapi/streaming/subscription-spec.js

```diff
@@ -112,6 +112,52 @@ describe("openapi StreamingSubscription", () => {
                 done();
             });
         });
+
+        it("retries subscribing if receive gateway timeout", (done) => {
+            var subscription = new Subscription('123', transport, 'serviceGroup', 'test/resource', {}, createdSpy, null, errorSpy);
+            subscription.onSubscribe();
+
+            transport.postReject({ status: "504", response: { message: "Gateway Timeout"}});
+
+            tick(() => {
+                expect(transport.delete.calls.count()).toEqual(1);
+                transport.deleteResolve();
+                tick(() => {
+
+                    jasmine.clock().tick(1000);
+                    // second time round server responds with 200
+                    sendInitialResponse();
+
+                    tick(() => {
+                        expect(errorSpy).not.toHaveBeenCalled();
+                        expect(createdSpy.calls.count()).toEqual(1);
+                        done();
+                    });
+                });
+            });
+        });
+
+        it("retries subscribing 5 times before giving up", (done) => {
```

Comment:
I think we need more tests for edge cases. 
* Unsubscribe request in the middle of retry.
* Modify request in the middle of retry.
* Subscribe request in the middle of retry. 
* Dispose (reset) request in the middle of retry.
* Modify which has timing out unsubscribe. But needs to follow up with subscribe afterwards (from queue).